src/Elliptic/Utilities/ApplyAt.hpp

```diff
@@ -0,0 +1,210 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Utilities to retrieve values from maps in tagged containers
+
+#pragma once
+
+#include <tuple>
+#include <utility>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Utilities/ForceInline.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+#include "Utilities/TupleSlice.hpp"
+
+namespace elliptic::util {
+namespace detail {
+
+template <bool PassThrough, typename... MapKeys>
+struct unmap_arg;
+
+template <typename... MapKeys>
+struct unmap_arg<true, MapKeys...> {
+  template <typename T>
+  static constexpr const T& apply(
+      const T& arg, const std::tuple<MapKeys...>& /*map_keys*/) noexcept {
+    return arg;
+  }
+
+  template <typename T>
+  static constexpr gsl::not_null<T*> apply(
+      const gsl::not_null<T*> arg,
+      const std::tuple<MapKeys...>& /*map_keys*/) noexcept {
+    return arg;
+  }
+};
+
+template <typename FirstMapKey, typename... MapKeys>
+struct unmap_arg<false, FirstMapKey, MapKeys...> {
+  template <typename T>
+  static constexpr auto apply(
```

Comment:
are you sure this (and below) sholudn't be a `decltype(auto)`?

Reply:
added `decltype`s. `DataBox` might be missing some as well